File: client/database/src/main/java/hr/foi/air/database/entities/Event.java
package hr.foi.air.database.entities;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverters;

import java.util.Date;

import hr.foi.air.database.converters.DateConverter;

@Entity(tableName = "events")
@TypeConverters(DateConverter.class)
public class Event {
    @PrimaryKey(autoGenerate = true)
    int id;

    String title;
    String description;

    @ColumnInfo(name = "creation_date")
    Date creationDate;
    @ColumnInfo(name = "start_time")
    Date startTime;
    @ColumnInfo(name = "location_addr")
    String address;
    @ColumnInfo(name = "location_lon")
    Double longitude;
    @ColumnInfo(name = "location_lat")
    Double latitude;
    @ForeignKey(entity = User.class, parentColumns = "id", childColumns = "creator_id")
    @ColumnInfo(index = true, name = "creator_id")
    int creatorId;
    @ForeignKey(entity = Sport.class, parentColumns = "id", childColumns = "sport_id")
    @ColumnInfo(index = true, name = "sport_id")
    int sportId;

    public int getSportId() {
        return sportId;
    }

    public void setSportId(int sportId) {
        this.sportId = sportId;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(Date creationDate) {
        this.creationDate = creationDate;
    }

    public Date getStartTime() {
        return startTime;
    }

    public void setStartTime(Date startTime) {
        this.startTime = startTime;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public Double getLongitude() {
        return longitude;
    }

    public void setLongitude(Double longitude) {
        this.longitude = longitude;
    }

    public Double getLatitude() {
        return latitude;
    }

    public void setLatitude(Double latitude) {
        this.latitude = latitude;
    }

    public int getCreatorId() {
        return creatorId;
    }

    public void setCreatorId(int creatorId) {
        this.creatorId = creatorId;
    }
}

File: server/database/entities/Event.php
<?php

class Event {
    private $id;
    private $title;
    private $description;
    private $creationDate;
    private $locationLon;
    private $locationLat;
    private $locationAddr;
    private $startTime;
    private $creatorId;
    private $sportId;
    
    public function getId() {
        return $this->id;
    }

    public function getTitle() {
        return $this->title;
    }

    public function getDescription() {
        return $this->description;
    }

    public function getCreationDate() {
        return $this->creationDate;
    }

    public function getLocationLon() {
        return $this->locationLon;
    }

    public function getLocationLat() {
        return $this->locationLat;
    }

    public function getLocationAddr() {
        return $this->locationAddr;
    }

    public function getStartTime() {
        return $this->startTime;
    }

    public function getCreatorId() {
        return $this->creatorId;
    }

    public function getSportId() {
        return $this->sportId;
    }

    public function setId($id): void {
        $this->id = $id;
    }

    public function setTitle($title): void {
        $this->title = $title;
    }

    public function setDescription($description): void {
        $this->description = $description;
    }

    public function setCreationDate($creationDate): void {
        $this->creationDate = $creationDate;
    }

    public function setLocationLon($locationLon): void {
        $this->locationLon = $locationLon;
    }

    public function setLocationLat($locationLat): void {
        $this->locationLat = $locationLat;
    }

    public function setLocationAddr($locationAddr): void {
        $this->locationAddr = $locationAddr;
    }

    public function setStartTime($startTime): void {
        $this->startTime = $startTime;
    }

    public function setCreatorId($creatorId): void {
        $this->creatorId = $creatorId;
    }

    public function setSportId($sportId): void {
        $this->sportId = $sportId;
    }
    
    public static function fromObject($obj) {
        $event = new Event();
        if (isset($obj->id)) {
            $event->id = $obj->id;
        }
        if (isset($obj->title)) {
            $event->title = $obj->title;
        }
        if (isset($obj->description)) {
            $event->description = $obj->description;
        }
        if (isset($obj->creationDate)) {
            $event->creationDate = $obj->creationDate;
        }
        if (isset($obj->locationLon)) {
            $event->locationLon = $obj->locationLon;
        }
        if (isset($obj->locationLat)) {
            $event->locationLat = $obj->locationLat;
        }
        if (isset($obj->locationAddr)) {
            $event->locationAddr = $obj->locationAddr;
        }
        if (isset($obj->startTime)) {
            $event->startTime = $obj->startTime;
        }
        if (isset($obj->creatorId)) {
            $event->creatorId = $obj->creatorId;
        }
        if (isset($obj->sportId)) {
            $event->sportId = $obj->sportId;
        }
        return $event;
    }

}